File: Data Structures & Algorithms/max-water-container/submission-1.java
class Solution {
    public int maxArea(int[] heights) {
        int ans = Integer.MIN_VALUE;
        int lo=0, hi = heights.length - 1;
        
        while(lo < hi) {
            int breadth = hi - lo;
            int length = Math.min(heights[lo], heights[hi]);
            int area = breadth * length;
            ans=Math.max(ans, area);
            if (heights[lo] <= heights[hi]) {
                lo++;
            } else {
                hi--;
            }
        }
        
        return ans;
    }
}

File: Data Structures & Algorithms/longest-increasing-subsequence/submission-1.py
class Solution:
    def lengthOfLIS(self, nums: List[int]) -> int:
        N = len(nums)
        ans=[1]*(N)
        for i in range(N):
            for k in range(i):
                if nums[i] > nums[k]:
                    ans[i] = max(ans[i], ans[k] + 1)
        # print(ans)
        return max(ans)